pkg/planner/core/task.go

```diff
@@ -674,10 +674,12 @@ func (p *PhysicalLimit) Attach2Task(tasks ...base.Task) base.Task {
 		} else if !cop.idxMergeIsIntersection {
 			// We only support push part of the order prop down to index merge build case.
 			if len(cop.rootTaskConds) == 0 {
-				if cop.indexPlanFinished {
-					// when the index plan is finished, sink the limit to the index merge table side.
+				// since issues/52947, we shouldn't push down part limit down to table plans, because
+				// the handles we pruned/limited is not according to index-order (internal handles reorder should be considered)
```

Comment:
<img width="624" alt="image" src="https://github.com/pingcap/tidb/assets/7846227/edca2e45-2e91-4ae0-9aca-5178d16ed40e">
We can follow the same comment like the double read case